packages/beacon-node/src/network/discv5/index.ts

```
@@ -73,18 +73,14 @@ export class Discv5Worker extends (EventEmitter as {new (): StrictEventEmitter<E
 
   onDiscoveredStr(enrBuf: Uint8Array): void {
     const enr = this.decodeEnr(enrBuf);
-    if (enr !== undefined) {
+    if (enr) {
       this.emit("discovered", enr);
     }
   }
 
   async enr(): Promise<ENR> {
     if (this.status.status === "started") {
-      const enr = this.decodeEnr(await this.status.workerApi.enrBuf());
-      if (enr === undefined) {
-        throw new Error("unable to decode self ENR");
-      }
-      return enr;
+      return ENR.decode(Buffer.from(await this.status.workerApi.enrBuf()));
```

Our own ENR is not untrusted input, so should throw an error if invalid